helper/builtinplugins/registry.go

```diff
@@ -144,34 +151,40 @@ func newRegistry() *registry {
 			"aws":      {Factory: logicalAws.Factory},
 			"azure":    {Factory: logicalAzure.Factory},
 			"cassandra": {
-				Factory:           logicalCass.Factory,
-				DeprecationStatus: consts.PendingRemoval,
+				Factory:           removedFactory,
+				DeprecationStatus: consts.Removed,
 			},
 			"consul":     {Factory: logicalConsul.Factory},
 			"gcp":        {Factory: logicalGcp.Factory},
 			"gcpkms":     {Factory: logicalGcpKms.Factory},
 			"kubernetes": {Factory: logicalKube.Factory},
 			"kv":         {Factory: logicalKv.Factory},
 			"mongodb": {
-				Factory:           logicalMongo.Factory,
+				Factory:           removedFactory,
+				DeprecationStatus: consts.Removed,
+			},
+			// The mongodbatlas logical database plugin was mistakenly left
+			// out of the deprecation process last release, so it will lag
+			// behind by one release cycle.
+			"mongodbatlas": {
+				Factory:           logicalMongoAtlas.Factory,
```

Comment:
This was not by mistake. MongoDB Atlas Secrets Engine and MongoDB Atlas DB Engines are two distinct plugins. The former is for managing API keys programmatically, and the latter is for managing DB users. We should not deprecate or remove this backend.